fdbserver/VersionedBTree.actor.cpp

```diff
@@ -5825,9 +5827,11 @@ class KeyValueStoreRedwoodUnversioned : public IKeyValueStore {
 				// Read page contents without using waits
 				bool isRoot = cur.inRoot();
 				BTreePage::BinaryTree::Cursor leafCursor = cur.popPath();
+				// we can bypass the bounds check for each key in the leaf if the entire leaf is in range
+				bool boundsCheck = leafCursor.lowerBound() < keys.begin;
```

Comment:
This can be <= because if the lower bound == the query begin then the lowest record in the page, even if equivalent in key to the lower bound, would still be included in the result set.

Reply:
Wouldn't that mean that we still want <? 
It short circuits the per-key comparisons only when boundsCheck = false. So when lower bound == query begin, we know we don't need to check the individual keys, since the lower bound is inclusive. So we only need to check the individual keys in the page when lower bound < query begin, so I think this is right.
And since upperBound is exclusive, along with keys.end, you should be able to also skip the page when upper bound == query end, and only have to check the individual keys in the page when upper bound > query end.
If the naming of "bounds check" was confusing I can try to think of a better name that encapsulates this behavior better.

Reply:
Yeah, I completely misinterpreted this despite the comments.
I think `doBoundsCheck` would be a better name, as I took `boundsCheck` to mean "the result of the boundary check is known and true" initially.